crates/pint-deploy/src/main.rs

```diff
@@ -87,14 +85,22 @@ async fn run(args: Args) -> anyhow::Result<()> {
             let contract_path = profile_dir.join(&pinned.name).with_extension("json");
             let contract = contract_from_path(&contract_path).await?;
             print_deploying(&pinned.name, &contract);
-            let output = builder_client.deploy_contract(&contract).await?;
-            print_received(&output);
+            submit(contract, &builder_address).await?;
         }
     }
 
     Ok(())
 }
 
+async fn submit(contract: Contract, builder_address: &String) -> Result<(), anyhow::Error> {
+    let registry_predicate = essential_node_types::BigBang::default().contract_registry;
+    let solution = register_contract_solution(registry_predicate, &contract)?;
```

Comment:
I'm considering whether it would be cleaner to move this stuff to the `pint-submit` lib, and just pass the contract in as an arg via the `submit_solution` function.